subsys/usb/device/usb_device.c

```diff
@@ -1626,8 +1629,16 @@ int usb_enable(usb_dc_status_callback status_cb)
 		goto out;
 	}
 
+	if (dev_desc->bDescriptorType != USB_DESC_DEVICE ||
```

Comment:
I would rather change this to assert because it indicates something seriously wrong in the device stack itself.